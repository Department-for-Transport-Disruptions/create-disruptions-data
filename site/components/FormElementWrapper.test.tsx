import * as React from "react";
import renderer from "react-test-renderer";
import FormElementWrapper from "./FormElementWrapper";
import { describe, it, expect } from "vitest";

describe("FormElementWrapper", () => {
    it("should render correctly with no errors", () => {
        const tree = renderer
            .create(<FormElementWrapper errors={[]} errorId={""} errorClass={""} children={<div></div>} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("should render correctly around an input with errors", () => {
        const tree = renderer
            .create(
                <FormElementWrapper
                    errors={[
                        {
                            errorMessage: "There was an error",
                            id: "test-div",
                        },
                    ]}
                    errorId={"test-div"}
                    errorClass={"govuk-input--error"}
                    children={<input id="test-div"></input>}
                />,
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
