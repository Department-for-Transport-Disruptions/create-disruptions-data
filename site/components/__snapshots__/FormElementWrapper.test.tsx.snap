// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FormElementWrapper > should render correctly around an input with errors 1`] = `
<div
  className=""
>
  <span
    className="govuk-error-message"
    id="test-div-error"
  >
    <span
      className="govuk-visually-hidden"
    >
      Error: 
    </span>
    There was an error
  </span>
  <input
    aria-describedby="test-div-error"
    className="govuk-input--error"
    id="test-div"
  />
</div>
`;

exports[`FormElementWrapper > should render correctly with no errors 1`] = `
<div
  className=""
>
  <div />
</div>
`;
